zone/entity.cpp

```diff
@@ -1200,6 +1200,52 @@ bool EntityList::IsMobSpawnedByNpcTypeID(uint32 get_id)
 	return false;
 }
 
+bool EntityList::IsNPCSpawned(std::vector<uint32> npc_ids)
+{
+	bool is_spawned = false;
+	if (npc_list.empty() || npc_ids.empty()) {
+		return is_spawned;
+	}
+
+	for (auto current_npc : npc_list) {
+		if (
+			std::find(
+				npc_ids.begin(),
+				npc_ids.end(),
+				current_npc.second->GetNPCTypeID()
+			) != npc_ids.end() && 
+			current_npc.second->GetID() != 0
+			) {
+			is_spawned = true;
```

Comment:
Can just break early here on the first match

Reply:
Alternatively to avoid code reuse it could just call the function below `return CountSpawnedNPCs() != 0` I think